Library/Homebrew/cmd/shellenv.sh

```diff
@@ -9,30 +9,56 @@
 # HOMEBREW_REPOSITORY is set by bin/brew
 # shellcheck disable=SC2154
 homebrew-shellenv() {
+  # Unordered subset check
+  # Usage: contains SEP VARNAME [ENTRY ...]
+  # Examples:
+  #   - contains ':' VAR aa bb cc  # unordered subset check
+  #   - contains ':' VAR aa:bb:cc  # ordered substring check (not subsequence)
+  contains() {
```

Comment:
Potentially simpler solution: `export HOMEBREW_SHELLENV=1` or similar and just make this whole command a no-op if that's already in the environment.

Reply:
@XuehaiPan I don't consider this resolved (yet). What are your thoughts on using a `HOMEBREW_SHELLENV` variable in this way? It would remove the need for the entire `contains` function and all the shell-conditional logic.

Reply:
> Potentially simpler solution: export `HOMEBREW_SHELLENV=1`

Do you mean set only one variable `HOMEBREW_SHELLENV` for all `PATH`, `MANPATH`, `INFOPATH`, not one for each?

> It would remove the need for the entire `contains` function and all the shell-conditional logic.

This function is used three times only up to now. We can replace it with three conditions. I have extracted the reusable function `contains` and hope we can add new features in the future. For example, make `brew shellenv` add brewed library to `LD_LIBRARY_PATH` / `C_INCLUDE_PATH` / `CPPFLAGS` for compilers, add completion path to `FPATH`, etc.

Reply:
> Do you mean set only one variable `HOMEBREW_SHELLENV` for all `PATH`, `MANPATH`, `INFOPATH`, not one for each?

No, I mean make `brew shellenv` also output `export HOMEBREW_SHELLENV=1` (or similar) and then if `HOMEBREW_SHELLENV` has been set in the environment then skip the rest of the execution rather than checking each individual entry.

> I don't know how to add warm-ups to performance tests :(

Use `hyperfine`

Reply:
> No, I mean make brew shellenv also output export HOMEBREW_SHELLENV=1 (or similar) and then if HOMEBREW_SHELLENV has been set in the environment then skip the rest of the execution rather than checking each individual entry.

Make sense. Looks much simpler, thanks.